### PR TITLE
[WIP] gravel/glass: service remove

### PR DESCRIPTION
**Pull request by s0nea**
Signed-off-by: Tatjana Dehler <tdehler@suse.com>

## Checklist
- [ ] References issues, create if required
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test tumbleweed`
- `jenkins run tumbleweed`
- `jenkins test leap`
- `jenkins run leap`

</details>

### CODE REVIEW

File: src/glass/src/app/pages/services-page/services-page.component.ts [hunk 1]
```diff
@@ -215,4 +228,29 @@ export class ServicesPageComponent {
       }
     });
   }
+
+  private removeService(serviceName: string): void {
+    this.dialogService.open(
+      DialogComponent,
+      (res: boolean) => {
+        if (res) {
+          this.blockUI.start(translate(TEXT('Please wait, deleting service ...')));
```
Comment (votdev) [inline src/glass/src/app/pages/services-page/services-page.component.ts:237]: If the action menu is `Remove service`, i suggest to use this wording in the waiting spinner too.
```suggestion
          this.blockUI.start(translate(TEXT('Please wait, removing service ...')));
```

File: src/glass/src/app/pages/services-page/services-page.component.ts [hunk 2]
```diff
@@ -215,4 +228,29 @@ export class ServicesPageComponent {
...
+              this.loadData();
+            });
+        }
+      },
+      {
+        width: '40%',
+        data: {
+          type: 'yesNo',
+          icon: 'question',
+          message: TEXT(`Do you really want to delete the service <strong>${serviceName}</strong>?`)
```
Comment (votdev) [inline src/glass/src/app/pages/services-page/services-page.component.ts:251]:
```suggestion
          message: TEXT(`Do you really want to remove the service <strong>${serviceName}</strong>?`)
```

File: src/glass/src/app/pages/services-page/services-page.component.ts [hunk 3]
```diff
@@ -151,6 +158,12 @@ export class ServicesPageComponent {
                 this.showMountCmdDialog(cmdArgs);
               });
             }
+          },
+          {
+            title: TEXT('Remove service'),
```
Comment (votdev) [inline src/glass/src/app/pages/services-page/services-page.component.ts:163]: We need to decide if we want to name it `delete` or `remove`. Any preferences?

File: src/gravel/api/services.py [hunk 4]
```diff
@@ -149,6 +149,20 @@ async def create_service(
     return CreateResponse(success=True)
 
 
+@router.delete("/{service_name}",
+               name="Delete service by name")
+async def delete(service_name: str, request: Request):
```
Comment (votdev) [inline src/gravel/api/services.py:154]: The code to secure access via authentication is missing.
```suggestion
async def delete(service_name: str, request: Request, _=Depends(jwt_auth_scheme)):
```

**Review by votdev — CHANGES_REQUESTED** [review]

File: src/glass/src/app/pages/services-page/services-page.component.ts (hunk 3 above)
Comment (s0nea) [reply src/glass/src/app/pages/services-page/services-page.component.ts:163]: I'm personally in favor of `remove`. But I can't tell the reason. It's more of a "feeling". My "feeling" might be wrong, e.g. we're already using `delete` at some places.

**Comment by s0nea** [comment]
I'm closing the pull request in favor of project target reassessing for now. Feel free to reopen it once it's relevant again.

**Pull request closed without merging**